src/envs/base_env.py

```diff
@@ -81,3 +115,24 @@ def seed(self, seed: int) -> None:
         self._train_rng = np.random.default_rng(self._seed)
         self._test_rng = np.random.default_rng(self._seed +
                                                CFG.test_env_seed_offset)
+
+    def reset(self, train_or_test: str, task_idx: int) -> State:
+        """Resets the current state to the train or test task initial state."""
+        task = self.get_task(train_or_test, task_idx)
+        self._current_state = task.init
+        # Copy to prevent external changes to the environment's state.
+        return self._current_state.copy()
+
+    def step(self, action: Action) -> State:
+        """Apply the action, and update and return the current state.
+
+        Note that this action is a low-level action (i.e., its array
+        representation is a member of self.action_space), NOT an option.
+
+        By default, this function just calls self.simulate. However,
+        environments that maintain a more complicated internal state may
```

Comment:
that maintain a more complicated internal state, or that don't implement simulate()*